### ISSUE TITLE
Watchman should not be looked for unless needed.

### ISSUE THREAD

**Issue by rwjblue** [ember-cli/ember-cli#2642]
It seems that we are initializing `watchman` before any invocation when most do not need watching.

A few examples:

```
% ember version

version: 0.1.3-master-0dc44fb0a8
valid watchman found, version: [3.0.0]
node: 0.10.33
npm: 2.1.10
```

```
% ember build
version: 0.1.3-master-0dc44fb0a8
valid watchman found, version: [3.0.0]
Built project successfully. Stored in "dist/".
```

```
% ember generate component foo-bar
version: 0.1.3-master-0dc44fb0a8
valid watchman found, version: [3.0.0]
installing
  create app/components/foo-bar.js
  create app/templates/components/foo-bar.hbs
installing
  create tests/unit/components/foo-bar-test.js
```

We only use file watching for a two commands: `serve` and `test` (as far as I can tell).  We should start watchman in them instead of all the time.

**Comment by szimek** [comment]
I'm currently working on Heroku buildpack for Ember CLI and it displays:
`Could not find watchman, falling back to NodeWatcher for file system events`
when it runs `ember build --environment=production` command (*without* `--watch` option) and I can't even pass `--watcher` option to get rid of it.

**Comment by stefanpenner** [comment]
@szimek doesn't seem like its a blocker thought right? Also im curious why you need a heroku build-pack, just deploy the static files in `dist/*` Seems several orders of magnitude simpler.

**Comment by szimek** [comment]
@stefanpenner Sure, it's not a blocker - I just mentioned it as one of cases where it may be annoying. 

Regarding the buildpack - you're right that it might be much simpler to just upload `dist/*` directory, but I still didn't have time to check out how to build Heroku slugs locally and upload them :smile: That's next on my todo list. The buildpack is [really, really simple](https://github.com/szimek/heroku-buildpack-ember-cli-without-webserver/blob/master/bin/compile#L37-L48) as well - all it does is install Bower packages and run `ember build --environmnt=production`.

**Comment by lindem** [comment]
Is there any hope of just permanently disabling these messages for people who don't want to install watchman (like a configuration setting or something)? I think they are very irritating even when I know what watchman is. I could alias --watcher=node into the command, but everyone on the project will have to do that, too, then. Sorry, it's just unneeded.

**Comment by stefanpenner** [comment]
not really a bug, if someone wants to fix i welcome it.

**Comment by stefanpenner** [comment]
> I think they are very irritating even when I know what watchman is. I could alias --watcher=node into the command, but everyone on the project will have to do that, too, then. Sorry, it's just unneeded.

you are welcome to submit a PR that enables this via the .ember-cli config

**Comment by stefanpenner** [comment]
fixed by #2642
